### PR TITLE
PostCSS 8 support

### PR DESCRIPTION
**Pull request by C-Lodder**

### CODE REVIEW

**Comment by C-Lodder** [comment]
@ai @XhmikosR Trying to add support for PostCSS 8.x here, but seem to be having issues with the source maps on Windows.
The tests run fine on Ubuntu as you can see from: https://github.com/C-Lodder/grunt-postcss/pull/31/checks?check_run_id=1137192879 however not locally.

### Received:
```
'{"version":3,"sources":["../../test/fixtures/a.css"],"names":[],"mappings":"AAAA,EACI,UACJ","file":"..\\\\mapAnnotation Path.css","sourcesContent":["a {\\n    color: #fff;\\n}\\n"]}'
```

### Expected:
```
'{"version":3,"sources":["../../test/fixtures/a.css"],"names":[ ],"mappings":"AAAA,EACI,UACJ","file":"../mapAnnotationPath.css","sourcesContent":["a {\\n    color: #fff;\\n}\\n"]}'
```

**Comment by nDmitry** [comment]
Thanks for the PR. I don't think dropping support for all node versions except the current is the right move.
It should be in sync with PostCSS: https://github.com/postcss/postcss/blob/master/package.json#L6

**Comment by C-Lodder** [comment]
@nDmitry oops my bad. Updated the PR.

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "repository": "C-Lodder/grunt-postcss",
   "license": "MIT",
   "engines": {
-    "node": ">= 8"
+    "node": ">= 14"
```
Comment (XhmikosR) [inline package.json:12]: This seems also wrong?

**Comment by XhmikosR** [comment]
About the Windows issue, if you modify the sourcemaps here, make sure you don't use the generic path functions; instead use the `path.posix` variants.

File: package.json (hunk 1 above)
Comment (C-Lodder) [reply package.json]: fixed, thanks

**Comment by XhmikosR** [comment]
Since I'm on Windows, I'll have a look today. Sourcemaps can be tricky because the paths should always be Unix ones (not sure what the specs say, but this works everywhere AFAICT from all these years).

**Comment by XhmikosR** [comment]
This turns out to be tricky.

We should move with postcss 8.x and make a new issue about the Windows issue so that we try to fix it later.

**Comment by C-Lodder** [comment]
I've managed to find a fix in the postcss core. Will submit a PR and see what the author has to say

**Pull request merged**